aten/src/ATen/native/cuda/Normalization.cuh

```diff
@@ -622,6 +624,14 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_backward_cuda_template(const Tenso
 
 template<typename scalar_t, typename index_t>
 std::tuple<Tensor, Tensor> batch_norm_stats_cuda_template(const Tensor& input_, double epsilon) {
+  if (input_.size(0) == 0) {  // empty batch
+    auto input_options = input_.options();
+    int64_t n_input = input_.size(1);
+    return std::make_tuple(
+      at::zeros({n_input}, input_options),
```

Comment:
Have we tested it with fp16 inputs?
Looks a little scary here as we are returning stats with pre-promoted data type. -> Later in our all_gather, we'll have inconsistent data type across processes. IDK if it's safe to do so.

We can move the promotion up and use the updated `input_options` to be on the safe side.

Reply:
Thanks! I have updated the code to return promoted data type. I have also tested that with half precision.